src/main/java/org/axonframework/springboot/nativex/AxonRuntimeHints.java

```diff
@@ -0,0 +1,238 @@
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider;
+import org.springframework.core.type.filter.AnnotationTypeFilter;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.lang.reflect.Method;
+import java.sql.Connection;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * This class is used to set runtime hints for axon applications. It finds the root package, and starts from there to
+ * search vor axon classes, and add reflection hints. From those classes it tries to find all the api classes, which
+ * need to have there constructors and methods available for serialisation. It does a few additional things which are
+ * needed to make it work.
+ *
+ * @author Gerard Klijs
+ * @since 4.8.0
+ */
+public class AxonRuntimeHints implements RuntimeHintsRegistrar {
+
+    private static final String PACKAGE_SEPARATOR = ".";
+    private static final String FOLDER_SEPARATOR = "/";
+
+    @Override
+    public void registerHints(RuntimeHints hints, ClassLoader classLoader) {
+        ReflectionHints reflectionHints = hints.reflection();
+        registerGrpcHints(reflectionHints);
+        List<Class<?>> axonClasses = getAxonClasses(classLoader);
+        registerReflectionHints(axonClasses, reflectionHints);
+        Set<Class<?>> apiClasses = new HashSet<>(axonSerializableClasses());
+        addClassesUsedInAggregateConstructors(apiClasses, axonClasses);
+        addClassesUsedInHandlers(apiClasses, axonClasses);
+        apiClasses.forEach(c -> registerForSerialisation(reflectionHints, c));
+        hints.resources().registerPattern("axonserver_download.txt");
+        hints.resources().registerPattern("SQLErrorCode.properties");
+        hints.proxies().registerJdkProxy(Connection.class);
+        hints.proxies().registerJdkProxy(TypeReference.of(
+                "org.axonframework.common.jdbc.UnitOfWorkAwareConnectionProviderWrapper$UoWAttachedConnection"));
+    }
+
+    private void registerGrpcHints(ReflectionHints hints) {
+        hints.registerType(EpollChannelOption.class, MemberCategory.DECLARED_FIELDS);
+    }
+
+    private List<Class<?>> getAxonClasses(ClassLoader classLoader) {
+        return getClasses(classLoader, getRootPackage())
+                .stream()
+                .filter(this::isAxonClass)
+                .toList();
+    }
+
+    private void registerReflectionHints(List<Class<?>> axonClasses, ReflectionHints hints) {
+        hints
+                .registerTypes(axonClasses.stream().map(TypeReference::of).toList(),
+                               TypeHint.builtWith(MemberCategory.PUBLIC_CLASSES,
```

Comment:
I assume the list of `MemberCategory` entries here is similar to the `TypeAccess.values()` we used earlier?

Reply:
Yes